archive/zip_archiver.go

```diff
@@ -83,12 +98,25 @@ func (a *ZipArchiver) ArchiveDir(indirname string) error {
 	defer a.close()
 
 	return filepath.Walk(indirname, func(path string, info os.FileInfo, err error) error {
+		tmpFilePath, _ := filepath.Abs(path)
+		tmpLongFileName, _ := filepath.Abs(indirname)
+		shortFileName := strings.TrimPrefix(tmpFilePath, tmpLongFileName)
```

Comment:
Please use `filepath.Rel` (it's already used below, you could just move that line up)

Reply:
I did try to use filepath.Rel at first this causes some problems when matching. The reason is that Rel path could look like this
/somedir/moredir/evenmoredir/../hi/
When trying to remove the prefix from that the prefix would just be
/somedir/moredir/hi/